platform/engineering/design-docs/2020-12-10-auto-deploy-cms-build.md

```diff
@@ -0,0 +1,278 @@
+
+#### Pain points
+
+- Content writers want to quickly draft, publish, and deploy content.
+  - However, `vets-website` is only deployed to production once per day.
+  - The CMS team can run content-only deploys independently of the daily production deploy, but that is triggered manually.
+- The content-only deploy does not include automated accessibility checks.
+- In local development, engineers can encounter confusing build errors that are due to the mismatch in outdated locally cached content and updated Liquid templates.
+
+### High Level Design
+
+There will be a new content-only auto-deploy job in addition to the existing full auto-deploy.
+- It will build content with CMS export data.
+- It will be a recurring job that executes on a more frequent schedule than the daily prod deploy.
+
+The schedule will run between 8am and 8pm ET to align with CMS support hours.
+- It will initially be set to run hourly as a conservative cadence.
+- It will then be incrementally adjusted to a higher frequency as the build system allows.
+
+To avoid deploying when content hasn't changed since the previous deploy, the job will compare the export data (e.g., using and archiving the checksums of the tarballs).
+
+To differentiate between the auto-deploys, we will refer to the new one as the **content-only auto-deploy** and the existing one as the **daily or full auto-deploy**.
+
+## Specifics
+
+### Detailed Design
+
+#### Creating a new job
+
+The content-only auto-deploy will be configured as a [seed job](https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/deployment/config/jenkins-vetsgov/seed_job.groovy) (e.g., `deploys/vets-gov-autodeploy-content`) in the `devops` repo. It should also be possible to manually trigger it.
+
+We will leverage the existing full auto-deploy job (`deploys/vets-gov-autodeploy-vets-website`) by having the content-only auto-deploy invoke it.
+- The `use_latest_release` flag will be set to `true` and the `release_wait` will be set to 0.
+- Since the content-only auto-deploy will be a recurring job, there are some concerns around potential conflicts with daily deploys.
+- Having the new job invoke `deploys/vetsgov-autodeploy-vets-website` will allow both auto-deploys to share a queue in Jenkins and avoid conflicts.
+- CMS stakeholders can still manually trigger the existing job for content-only deploys as before.
+
+The Jenkinsfile for the content-only deploy (e.g., `ansible/Jenkinsfiles/deploys/content-only-autodeploy`) would contain this fundamental step:
+```
+stage('Autodeploy content') {
+  steps {
+    build job: 'deploys/vets-gov-autodeploy-vets-website', parameters: [
+      stringParam(name: 'release_wait', value: '0'),
+      booleanParam(name: 'use_latest_release', value: true),
+    ], wait: true
+  }
+}
+```
```

Comment:
@dginther: Just updated this section and wanted to double check my assumptions about how Jenkins works.

Basically we want to be able to run the same `deploys/vets-gov-autodeploy-vets-website` job, but with different cron trigger and parameters.